tesseract_collision/test/collision_margin_data_unit.cpp

```diff
@@ -0,0 +1,241 @@
+    EXPECT_NEAR(data.getMaxCollisionMargin(), pair_margin, tol);
+    EXPECT_NEAR(data.getPairCollisionMarginData("link_1", "link_2"), pair_margin, tol);
+    EXPECT_EQ(data.getPairsCollisionMarginData().size(), 1);
+  }
+
+  {  // Test Apply Override Link Pair
+    double default_margin = 0.0254;
+    double pair_margin = 0.5;
+    CollisionMarginData data(default_margin);
+    data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+
+    pair_margin = pair_margin * 3;
+    CollisionMarginData override_data(default_margin * 3);
+    override_data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+    data.apply(override_data, CollisionMarginOverrideType::OVERRIDE_PAIR_MARGIN);
+    EXPECT_NEAR(data.getDefaultCollisionMarginData(), default_margin, tol);
+    EXPECT_NEAR(data.getMaxCollisionMargin(), pair_margin, tol);
+    EXPECT_NEAR(data.getPairCollisionMarginData("link_1", "link_2"), pair_margin, tol);
+    EXPECT_EQ(data.getPairsCollisionMarginData().size(), 1);
+  }
+
+  {  // Test Apply Override Replace
+    double default_margin = 0.0254;
+    double pair_margin = 0.5;
+    CollisionMarginData data(default_margin);
+    data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+
+    default_margin = default_margin * 3;
+    pair_margin = pair_margin * 3;
+    CollisionMarginData override_data(default_margin);
+    override_data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+    data.apply(override_data, CollisionMarginOverrideType::REPLACE);
+    EXPECT_NEAR(data.getDefaultCollisionMarginData(), default_margin, tol);
+    EXPECT_NEAR(data.getMaxCollisionMargin(), pair_margin, tol);
+    EXPECT_NEAR(data.getPairCollisionMarginData("link_1", "link_2"), pair_margin, tol);
+    EXPECT_EQ(data.getPairsCollisionMarginData().size(), 1);
+  }
+
+  {  // Test Apply Override None
+    double default_margin = 0.0254;
+    double pair_margin = 0.5;
+    CollisionMarginData data(default_margin);
+    data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+
+    CollisionMarginData override_data(default_margin * 3);
+    override_data.setPairCollisionMarginData("link_1", "link_2", pair_margin * 3);
+    data.apply(override_data, CollisionMarginOverrideType::NONE);
+    EXPECT_NEAR(data.getDefaultCollisionMarginData(), default_margin, tol);
+    EXPECT_NEAR(data.getMaxCollisionMargin(), pair_margin, tol);
+    EXPECT_NEAR(data.getPairCollisionMarginData("link_1", "link_2"), pair_margin, tol);
+    EXPECT_EQ(data.getPairsCollisionMarginData().size(), 1);
+  }
+
+  {  // Test Apply Override Modify Pair
+    double default_margin = 0.0254;
+    double pair_margin = 0.5;
+    CollisionMarginData data(default_margin);
+    data.setPairCollisionMarginData("link_1", "link_2", pair_margin);
+
+    CollisionMarginData override_data(default_margin * 3);
+    override_data.setPairCollisionMarginData("link_1", "link_3", pair_margin * 3);
```

Comment:
Suggest adding case where override has link pair already in data

Reply:
Done.